docs/specification.md

```diff
@@ -61,358 +61,973 @@ For a broader understanding of A2A's purpose and benefits, see [What is A2A?](./
+**Inputs:**
 
-### 4.5. In-Task Authentication (Secondary Credentials)
+- None (no parameters required)
 
-If an agent, during the execution of a task, requires _additional_ credentials for a _different_ system or resource (e.g., to access a specific tool on behalf of the user that requires its own auth):
+**Outputs:**
 
-1. It **SHOULD** transition the A2A task to the `auth-required` state (see [`TaskState`](#63-taskstate-enum)).
-2. The accompanying `TaskStatus.message` (often a [`DataPart`](#653-datapart-object)) **SHOULD** provide details about the required secondary authentication, potentially using an [`PushNotificationAuthenticationInfo`](#69-pushnotificationauthenticationinfo-object)-like structure to describe the need.
-3. The A2A Client then obtains these new credentials out-of-band and provides them in a subsequent [`message/send`](#71-messagesend) or [`message/stream`](#72-messagestream) request. How these credentials are used (e.g., passed as data within the A2A message if the agent is proxying, or used by the client to interact directly with the secondary system) depends on the specific scenario.
+- [`AgentCard`](#441-agentcard): A complete Agent Card object, which may contain additional details or skills not present in the public card
 
-### 4.6. Authorization
+**Behavior:**
 
-Once a client is authenticated, the A2A Server is responsible for authorizing the request based on the authenticated client/user identity and its own policies. Authorization logic is implementation-specific and MAY be enforced based on:
+- **Authentication**: The client MUST authenticate the request using one of the schemes declared in the public `AgentCard.securitySchemes` and `AgentCard.security` fields.
+- **Extended Information**: The operation MAY return different details based on client authentication level, including additional skills, capabilities, or configuration not available in the public Agent Card.
+- **Card Replacement**: Clients retrieving this authenticated card SHOULD replace their cached public Agent Card with the content received from this endpoint for the duration of their authenticated session or until the card's version changes.
+- **Availability**: This operation is only available if the public Agent Card declares `supportsAuthenticatedExtendedCard: true`.
 
-- The specific skills requested (e.g., as identified by `AgentSkill.id` from the Agent Card).
-- The actions attempted within the task.
-- Data access policies relevant to the resources the agent manages.
-- OAuth scopes associated with the presented token, if applicable.
+**Transport Implementations:**
 
-Servers should implement the principle of least privilege.
+- **JSON-RPC**: [`agent/getAuthenticatedExtendedCard`](#938-agentgetauthenticatedextendedcard)
+- **gRPC**: [`GetExtendedAgentCard`](#103-core-methods)
+- **HTTP/REST**: [`GET /v1/extendedAgentCard`](#1124-agent-card)
 
-## 5. Agent Discovery: The Agent Card
+#### 3.1.8. Set or Update Push Notification Config
```

Comment:
Seems weird that this method isn't grouped with the other task related operations (resubscribe etc), and the AgentCard operation is in-between. Might be worth re-ordering the spec slightly to make that make more sense, or even creating explicit groups of operations on objects, like GetTask, Resubscribe etc are all operations on a Task.